Scripts/RemediationScripts/Remediate-EnableTransparentDataEncryption.ps1

```diff
@@ -0,0 +1,653 @@
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 4] Preparing to fetch all SQL Servers..."
+
+    $sqlServersResources = @()
+    $databaseList = @()
+
+    # Includes SQL Servers where Transparent Data Encryption (TDE) is enabled.
+    $sqlDatabaseWithTDEEnabled = @()
+
+    # Includes SQL Servers where Transparent Data Encryption (TDE) is not enabled.
+    $sqlDatabaseWithTDEDisabled = @()
+
+    # Includes SQL Servers that were skipped during remediation. There were errors remediating them.
+    $sqlServersSkipped = @()
+
+    # No file path provided as input to the script. Fetch all SQL Servers in the Subscription.
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        Write-Host "Fetching all SQL Servers in Subscription: $($context.Subscription.SubscriptionId)" -ForegroundColor $([Constants]::MessageType.Info)
+
+        # Get all SQL Servers in a Subscription      
+        $sqlServersResources = Get-AzSqlServer
+        $totalSqlServers = ($sqlServersResources | Measure-Object).Count
+
+        if ($totalSqlServers -eq 0)
+        {
+            Write-Host "No SQL Server found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            break
+        }
+    
+        Write-Host "Found $($totalSqlServers) SQL Server(s)." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host "Fetching SQL Server configurations..."
+
+        
+
+        $sqlServersResources | ForEach-Object{
+            $databaseList = Get-AzSqlDatabase -ServerName $_.ServerName -ResourceGroupName $_.ResourceGroupName
+            $databaseList | ForEach-Object{
+                if ($_.DatabaseName -ne 'master')
+                {
+                    $tdeDetails = Get-AzSqlDatabaseTransparentDataEncryption -ServerName $_.ServerName -DatabaseName $_.DatabaseName -ResourceGroupName $_.ResourceGroupName
+                    if ($tdeDetails.State -eq "Enabled")
+                    {
+                        $sqlDatabaseWithTDEEnabled += $tdeDetails
+                    }
+                    else
+                    {
+                        $sqlDatabaseWithTDEDisabled += $tdeDetails
+                    }
+                }                
+            }
+        }
+    }
+    else
+    {
+        if (-not (Test-Path -Path $FilePath))
+        {
+            Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+
+        Write-Host "Fetching all SQL Servers from $($FilePath)" -ForegroundColor $([Constants]::MessageType.Info)
+
+        $sqlDatabaseWithTDEDisabled = Import-Csv -LiteralPath $FilePath
+    }
+
+    $totalSqlDatabaseWithTDEDisabled = ($sqlDatabaseWithTDEDisabled | Measure-Object).Count
```

Comment:
`Database` in `totalSqlDatabaseWithTDEDisabled` to be replaced with `Databases`. One of `Sql`/`TDE` to be renamed to be uniform with the other.